iotiletest/iotile/fixture/device_fixture.py

```diff
@@ -50,7 +50,7 @@ def device(port, device_id, direct, record, request):
         # before giving up to improve test robustness.
         max_attempts = 3
         i = 0
-        for i in range(0, max_attempts):
+        for i in range(0, max_attempts + 1):
```

Comment:
Curious about the logic here.  This change should make it try `max_attempts + 1` times instead of `max_attempts` times.  If `3` attempts isn't enough `max_attempts` should just be increased.

Reply:
The thing is that below, there is a check i == max_attempts, that is not possible if a exclusive range is used

Reply:
Ah, thanks for the clarification.  Can you just fix the below line to be `max_attempts - 1`?

Reply:
Oh course